guides/source/active_model_basics.md

```diff
@@ -44,76 +67,331 @@ class EmailContact
+attributes of a model. This module is particularly useful to simplify attribute
+access and manipulation, and it can add custom prefixes and suffixes to the
+methods of a class. You can define the prefixes and suffixes and which methods
+on the object will use them as follows:
+
+1. Include `ActiveModel::AttributeMethods` in your class.
+2. Call each of the methods you want to add, such as `attribute_method_suffix`,
+   `attribute_method_prefix`, `attribute_method_affix`.
+3. Call `define_attribute_methods` after the other methods to declare the
+   attribute(s) that should be prefixed and suffixed.
+4.  Define the various generic `_attribute` methods that you have declared. The
+    parameter `attribute` in these methods will be replaced by the argument
+    passed in `define_attribute_methods`. In our case it is `name`.
+
+NOTE: `attribute_method_prefix` and `attribute_method_suffix` are used to define
+the prefixes and suffixes that will be used to create the methods.
+`attribute_method_affix` is used to define both the prefix and suffix at the
+same time.
 
 ```ruby
 class Person
   include ActiveModel::AttributeMethods
 
-  attribute_method_prefix 'reset_'
-  attribute_method_suffix '_highest?'
-  define_attribute_methods 'age'
+  attribute_method_affix prefix: "reset_", suffix: "_to_default!"
+  attribute_method_prefix "first_"
+  attribute_method_prefix "last_"
+  attribute_method_suffix "_short?"
+
+  define_attribute_methods "name"
 
-  attr_accessor :age
+  attr_accessor :name
 
   private
-    def reset_attribute(attribute)
-      send("#{attribute}=", 0)
-    end
 
-    def attribute_highest?(attribute)
-      send(attribute) > 100
-    end
+  # this will create a method 'first_name'
+  def first_attribute(attribute)
+    public_send(attribute).split.first
+  end
+
+  # this will create a method 'last_name'
+  def last_attribute(attribute)
+    public_send(attribute).split.last
+  end
+
+  # this will create a method 'name_short?'
+  def attribute_short?(attribute)
+    public_send(attribute).length < 5
+  end
+
+  # this will create a method 'reset_name_to_default!'
+  def reset_attribute_to_default!(attribute)
+    public_send("#{attribute}=", "Default Name")
+  end
 end
 ```
 
 ```irb
 irb> person = Person.new
-irb> person.age = 110
-irb> person.age_highest?
-=> true
-irb> person.reset_age
-=> 0
-irb> person.age_highest?
+
+irb> person.name = "Jane Doe"
+
+# utilizing the `first_name` and `last_name` prefix methods
+irb> person.first_name
+=> "Jane"
+irb> person.last_name
+=> "Doe"
+
+# utilizing the `name_short?` suffix method
+irb> person.name_short?
 => false
+
+# utilizing the `reset_name_to_default!` prefix and suffix methods
+irb> person.reset_name_to_default!
+=> "Default Name"
+```
+
+If you call a method that is not defined, it will raise a `NoMethodError`
+error.
+
+#### Method: `alias_attribute`
+
+`ActiveModel::AttributeMethods` also provides aliasing of attribute methods.
```

Comment:
Again with aliasing: why would one want to do this? Is there anything we can add to explain this just a bit for beginners? I can clearly follow the example below to see how it works, but I don't get why one would want to do this. Especially in this example where the alias is longer than the original attribute name ('nickname' vs 'name').

Reply:
I've added the "why" to the alias info further up, but then I also added some more context specific descriptions here. 

I've also changed the example from 'name with an alias of nickname', to 'name with an alias of full_name' because it's more common to alias "name" with "full_name" to be more descriptive that the attribute includes the first name and  last name.